src/components/remote_control/src/resource_allocation_manager_impl.cc

```diff
@@ -0,0 +1,94 @@
+#include "remote_control/resource_allocation_manager_impl.h"
+#include "application_manager/application.h"
+
+namespace remote_control {
+
+CREATE_LOGGERPTR_GLOBAL(logger_, "RemoteControlModule")
+
+ResourceAlocationManagerImpl::ResourceAlocationManagerImpl(
+    application_manager::Service& service)
+    : service_(service) {}
+
+AquireResult::eType ResourceAlocationManagerImpl::AquireResource(
+    const std::string& module_type, uint32_t app_id) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  const application_manager::ApplicationSharedPtr aquiring_app =
+      service_.GetApplication(app_id);
+  if (!aquiring_app) {
+    LOG4CXX_WARN(logger_, "App with app_id " << app_id << "does not exist!");
+    return AquireResult::IN_USE;
+  }
+
+  const AllocatedResources::const_iterator allocated_it =
+      allocated_resources_.find(module_type);
+  if (allocated_resources_.end() == allocated_it) {
+    allocated_resources_[module_type] = app_id;
+    LOG4CXX_DEBUG(logger_,
+                  "Resource is not ys aquired. "
+                  "Allow "
+                      << app_id << " aquiring " << module_type);
+    return AquireResult::ALLOWED;
+  }
+
+  if (AccessMode::AUTO_DENY == current_access_mode_) {
```

Comment:
@AByzhynar if policy id AUTODENY but resource is not acquired , SDL should allow acquire this resource